webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java

```diff
@@ -82,22 +85,26 @@ public StudentGroupBatch upload(final User user, final MultipartFile file) {
                 .filename(file.getOriginalFilename());
 
         final MessageDigest messageDigest = DigestUtils.getMd5Digest();
-        final List<CsvValidationFailure> validationFailures = new ArrayList<>();
 
         try {
+            final List<CsvValidationFailure> validationFailures = newArrayList();
+            final Map<String, String> fileProperties = newHashMap();
+            fileProperties.put("username", user.getUsername());
+
             try (final DigestInputStream digestStream = new DigestInputStream(file.getInputStream(), messageDigest)) {
-                validationFailures.addAll(validate(digestStream, permissionScope));
+                validate(digestStream, permissionScope, validationFailures, fileProperties);
             }
 
             final String digest = Hex.encodeHexString(messageDigest.digest()).toUpperCase();
             if (validationFailures.isEmpty()) {
                 logger.debug("Validated request: {}", file.getOriginalFilename());
-                archiveFile(file, digest, user);
+                archiveFile(file, digest, fileProperties);
                 studentGroupBatchBuilder
                         .digest(digest)
                         .status(ImportStatus.ACCEPTED);
             } else {
-                logger.debug("Request failed validation: {}", file.getOriginalFilename());
+                logger.info("Request failed validation: {}", file.getOriginalFilename());
+                archiveFile(file, digest, fileProperties);
```

Comment:
sneaking in another change: archive the file even if the data is bad

Reply:
Nice!